ext/rugged/rugged_remote.c

```diff
@@ -181,6 +181,17 @@ void rugged_remote_init_callbacks_and_payload_from_options(
 	}
 }
 
+static int parse_prune_type(VALUE rb_prune_type)
+{
+	if (rb_prune_type == Qtrue) {
+		return GIT_FETCH_PRUNE;
+	} else if (rb_prune_type == Qfalse) {
+		return GIT_FETCH_NO_PRUNE;
+	} else {
+		return GIT_FETCH_PRUNE_UNSPECIFIED;
```

Comment:
`GIT_FETCH_PRUNE_UNSPECIFIED` is the default behaviour and it was intentional that I did it this way. If its preferred to type check and raise and error if the value is not `true / false / nil` then I will update it to work like that.